nbcelltests/test.py

```diff
@@ -12,41 +12,81 @@
 import sys
 import subprocess
 import tempfile
+import ast
 from .define import TestMessage, TestType
-from .shared import extract_cellsources, extract_celltests, extract_extrametadata
+from .shared import extract_extrametadata
 from .tests_vendored import BASE, JSON_CONFD
 
+# TODO: eventually want assemble() and the rest to be doing something
+# better than building up code in strings. It's tricky to work with,
+# tricky to read, can't lint, can't test easily, unlikely to be
+# generally correct, etc etc :)
```

Comment:
It is indeed hard to follow the code in the review ;)

Reply:
Although I added the TODO, this is not part of the current PR (was like this before the PR, and still like it after).